jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/CreateLambdaFunction.kt

```diff
@@ -63,12 +64,13 @@ class CreateLambdaFunction(
             return
         }
 
-        val templateFunctionHandlers = CloudFormationTemplateIndex.listFunctions(element.project)
-            .mapNotNull { it.handler() }
-            .toSet()
+        val project = e.getRequiredData(LangDataKeys.PROJECT)
 
-        val allowAction = lambdaHandlerResolver.determineHandlers(element, element.containingFile.virtualFile)
-            .none { it in templateFunctionHandlers }
+        val allowAction = lambdaHandlerResolver.determineHandler(element)?.let { handler ->
```

Comment:
Not a fan of this. It is implicit behavior that we can't explain why it is present sometimes and not others in the UI. It goes beyond the old logic of: Oh, it is because it is in a template to In template, yet not sam build supported.

Reply:
The current logic for whether to show this `CreateLambdaFunction` action already is implicit. Ie. if a handler is defined in a template, we don't show it, otherwise, we show it.

I agree this is making this logic more complicated and more implicit, and harder to maintain as well. But to follow the above principal, I think we should change the current logic to still show the `CreateLambdaFunction` action **ALL** the time and pop up an error/warning notification depending on the scenarios.

Thoughts?